### PR TITLE
Containers for local devcontainer and CI workflows

### PR DESCRIPTION
**Pull request by stonier**
# 🎉 New feature

Container workflows. Analogous to https://github.com/maliput/maliput/pull/574.

NB: Duplication discussion in https://github.com/maliput/maliput/issues/579

## Summary

Enables the local development workflow and unblocks CI bazel actions.

## Checklist
- [x] Signed all commits for DCO
- [x] Updated documentation (as needed)

### CODE REVIEW

File: .devcontainer/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+* Go to Codespaces
+* Select `New with Options`
+* Select `Bazel Zen` from the `Dev Container Configuration`
+
+<img src="./resources/codespaces.png" alt="codespaces" width="600">
+
+* Open a terminal in the container and run
+
+```
+@<github-username> ➜ /workspaces/maliput_malidrive (main) $ bazel build //...
```
Comment (francocipollone) [inline .devcontainer/README.md:47]: FTR, this isn't working yet until the repo is bazel-compatible

File: .devcontainer/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+Locally:
+
+* [Install VSCode](https://code.visualstudio.com/docs/setup/linux#_debian-and-ubuntu-based-distributions)
+* Open the project in VSCode
+* CTRL-SHIFT-P &rarr; Reopen in Container
+* Open a terminal in the container and run
+
+```
+(docker) zen@bazel-zen:/workspaces/maliput$ bazel build //...
```
Comment (francocipollone) [inline .devcontainer/README.md:33]: nit
```suggestion
(docker) zen@bazel-zen:/workspaces/maliput_malidrive$ bazel build //...
```

**Review by francocipollone — APPROVED** [review]
LGTM!

File: .devcontainer/README.md (hunk 1 above)
Comment (stonier) [reply .devcontainer/README.md:47]: Yes &rarr; PR #236

File: .devcontainer/README.md (hunk 2 above)
Comment (stonier) [reply .devcontainer/README.md:33]: Will get this in, rebase and fix in #236.

**Pull request merged**